### PR TITLE
Feat/new map solution

### PR DESCRIPTION
**Pull request by simon-the-shark**
So I did try to use flutter map package and imo it's pretty lightweight. Also it's kinda like building blocks so we can add more features later and we have very big control over what's happening there. Drawbacks to this is e.g. that we have to implement my location button ourselves.

So open street map doesn't look as cool as other providers, but is 100% free, pretty functional and if we want to use different style we can try to selfhost it at some point in the future. 

potencial TODOs for the future:
- map tile's caching with hive: https://pub.dev/packages/flutter_map_cache (imo quite URGENT)
- animations instead of instant camera jump: https://pub.dev/packages/flutter_map_animations (to consider)
- marker clustering: flutter_map_marker_cluster (to consider)
- all packages in the ecosystem list: https://github.com/josxha/awesome-flutter-map

tell me what you think :)
image

### CODE REVIEW

File: lib/features/buildings_view/model/building_model.dart [hunk 1]
```diff
@@ -19,8 +19,6 @@ class BuildingModel extends Building implements GoogleNavigable {
   @override
   LatLng get location => LatLng(latitude, longitude);
 
-  MarkerId get markerId => MarkerId(id);
-
   String? get addresFormatted =>
```
Comment (mikolaj-jalocha) [inline lib/features/buildings_view/model/building_model.dart:22]: typo here, missing "s" in "addres"

File: lib/features/map_view/controllers/map_controller.dart
Comment (mikolaj-jalocha) [inline lib/features/map_view/controllers/map_controller.dart:1]: Can the old animations be placed here? (The ones before changes)

File: lib/features/map_view/widgets/map_atrribution.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class OpenMapAtrribution extends ConsumerWidget {
+  const OpenMapAtrribution({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return RichAttributionWidget(
+      attributions: [
+        TextSourceAttribution(
+          "OpenStreetMap contributors",
+          onTap: () async => ref.launch("https://openstreetmap.org/copyright"),
```
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/map_atrribution.dart:17]: Maybe an overkill but what about extracting that url?

File: lib/features/map_view/widgets/map_atrribution.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+/// legal requirements for using OpenStreetMap
+class OpenMapAtrribution extends ConsumerWidget {
+  const OpenMapAtrribution({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return RichAttributionWidget(
+      attributions: [
+        TextSourceAttribution(
+          "OpenStreetMap contributors",
```
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/map_atrribution.dart:16]: Maybe an overkill but what about extracting that String?

File: lib/features/map_view/widgets/map_widget.dart [hunk 4]
```diff
@@ -1,61 +1,53 @@
...
+                .mapMoved(); // stop following location on user interaction
+          }
+        },
+        onTap:
+            ref.watch(context.mapController<T>().notifier).onMapBackgroundTap,
+      ),
+      mapController: ref.watch(context.mapController<T>()),
+      children: [
+        TileLayer(
+          urlTemplate: "https://tile.openstreetmap.org/{z}/{x}/{y}.png",
```
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/map_widget.dart:41]: What about extracting that?

File: lib/features/map_view/widgets/map_widget.dart
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/map_widget.dart:1]: What happened with old-school AsyncError, AsyncValue, AsyncLoading handling here? Just curious how we handle it now. Does FlutterMap has sth by default?

File: lib/features/map_view/widgets/map_widget.dart [hunk 5]
```diff
@@ -1,61 +1,53 @@
...
+          userAgentPackageName:
+              Platform.isIOS ? "com.solvro.ToPwr" : "com.solvro.topwr",
```
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/map_widget.dart:43]: Maybe extracting that as well?

File: lib/features/map_view/widgets/markers_layer.dart [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+class MarkersConsumerLayer<T extends GoogleNavigable> extends ConsumerWidget {
+  const MarkersConsumerLayer({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncItems = ref
+        .watch(context.mapSourceRepository<T>())
+        .valueOrNull
+        .whereNonNull
```
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/markers_layer.dart:17]: Looks like a good place to put some custom extension ;) Just suggestion

File: lib/features/my_location_button/presentation/my_loc_button.dart
Comment (mikolaj-jalocha) [inline lib/features/my_location_button/presentation/my_loc_button.dart:1]: Imo on Android Fab is to much to the top. Just mine opinion.

Comment (mikolaj-jalocha) [reply lib/features/my_location_button/presentation/my_loc_button.dart:1]: Maybe some padding or sth

File: lib/features/navigator/app_router.dart [hunk 7]
```diff
@@ -29,7 +29,7 @@ class _NoTransitionRoute extends CustomRoute {
 }
 
 @AutoRouterConfig(replaceInRouteName: "View,Route")
-class AppRouter extends _$AppRouter {
+class AppRouter extends RootStackRouter {
```
Comment (mikolaj-jalocha) [inline lib/features/navigator/app_router.dart:32]: Why the change here? Just curious

File: lib/features/parkings_view/parkings_view.dart [hunk 8]
```diff
@@ -27,15 +27,18 @@ class ParkingsView extends ConsumerWidget {
...
-        markerId: MarkerId(item.id),
-        position: item.location,
-        icon: isActive
-            ? MapMarkerUtils.activeParkingMapMarker
-            : MapMarkerUtils.parkingMapMarker,
-        onTap: () {
-          ref.read(parkingsMapControllerProvider.notifier).onMarkerTap(item);
-        },
+        alignment: Alignment.topCenter,
+        child: GestureDetector(
```
Comment (mikolaj-jalocha) [inline lib/features/parkings_view/parkings_view.dart:31]: Looks nice. Did you think about some splash effect here? Could improve UX

File: lib/features/splash_screen/splash_screen_controller.dart
Comment (mikolaj-jalocha) [inline lib/features/splash_screen/splash_screen_controller.dart:1]: Lack map utils here imo improves initial performance. Nice

File: lib/features/splash_screen/splash_screen_controller.dart [hunk 9]
```diff
@@ -8,27 +8,23 @@ import "package:riverpod_annotation/riverpod_annotation.dart";
...
-  Future<void> _initializationLogic(BuildContext context) async {
+  Future<void> _initializationLogic() async {
     /*
       Insert here any initialization async logic or operations 
       to be performed, when SplashScreen is showed.
     */
-    await MapMarkerUtils.loadMapMarkerAssets(context);
     await initHiveForGraphqlCache();
     await AppBarLogo.precacheImageIfAbsent();
-    await MapController.initializeGoogleMapsRenderingAndroid();
```
Comment (mikolaj-jalocha) [inline lib/features/splash_screen/splash_screen_controller.dart:26]: Especially line here ;)

File: lib/features/map_view/widgets/toolbar.dart
Comment (mikolaj-jalocha) [inline lib/features/map_view/widgets/toolbar.dart:1]: Can compass be seen on Android?

Comment (mikolaj-jalocha) [reply lib/features/map_view/widgets/toolbar.dart:1]: I can't see that 😂

**Review by mikolaj-jalocha — COMMENTED** [review]
Good work!  Substantial amount of code here've been refactored. Few notes below:

- In my opinion OSM design looks quite nice. It is very readable and for me - suits even better than previous  one. But interesting and challenging task for the future (regarding creating our own design)'d nice ;) 
- At least on emulator performance's been significantly improved
- I think there's need for good looking animation under the condition that it wouldn't reduce performance. 
- On Android device I couldn't see the compass 
- Below I posted some of my suggestion (minor ones). Treat them as suggestions, not required changes of any kind. 
- Also, I've posted some questions of mine regarding current solutions 
- I've created some tasks that you mentioned. Please convert them to issues, once pr's resolved ;)  Currently  I didn't include markers clustering, idk what are the pros and cons here. Explanation needed 😂

File: lib/features/map_view/controllers/map_controller.dart
Comment (simon-the-shark) [reply lib/features/map_view/controllers/map_controller.dart:1]: yeah but we need separate package for this to be able to animate camera, and the way it's structured, we need to do some refactor with this too: https://pub.dev/packages/flutter_map_animations

File: lib/features/map_view/widgets/map_atrribution.dart (hunk 2 above)
Comment (simon-the-shark) [reply lib/features/map_view/widgets/map_atrribution.dart:17]: cool, no problem

File: lib/features/navigator/app_router.dart (hunk 7 above)
Comment (simon-the-shark) [reply lib/features/navigator/app_router.dart:32]: it's caused by a new version of `auto_route`

File: lib/features/splash_screen/splash_screen_controller.dart
Comment (simon-the-shark) [reply lib/features/splash_screen/splash_screen_controller.dart:1]: super

File: lib/features/map_view/widgets/toolbar.dart
Comment (simon-the-shark) [reply lib/features/map_view/widgets/toolbar.dart:1]: it's disabled when the map is straight to the north. It should pop up when you rotate the map. It can be configured otherwhise too

File: lib/features/parkings_view/parkings_view.dart (hunk 8 above)
Comment (simon-the-shark) [reply lib/features/parkings_view/parkings_view.dart:31]: I like the idea

File: lib/features/map_view/widgets/markers_layer.dart (hunk 6 above)
Comment (simon-the-shark) [reply lib/features/map_view/widgets/markers_layer.dart:17]: this requires both `ref` and `context`, so imo won't be more readable

File: lib/features/my_location_button/presentation/my_loc_button.dart
Comment (simon-the-shark) [reply lib/features/my_location_button/presentation/my_loc_button.dart:1]: good idea

File: lib/features/map_view/widgets/map_widget.dart
Comment (simon-the-shark) [reply lib/features/map_view/widgets/map_widget.dart:1]: we previously had here async loading etc for permissions and now `IsFollowingCurrentLocationController` is not an async provider. It always has some boolean value, cause it's `build` returns `bool` and not `Future<bool>`

when it comes to flutter map, we're using it mechanism for following location, but it's not super readable

Comment (simon-the-shark) [reply lib/features/map_view/widgets/map_widget.dart:1]: actually I've tested it without permissions granted and it still asks us straight up when we open the map, so I'm gonna do a small adjustment here, thanks

File: lib/features/parkings_view/parkings_view.dart (hunk 8 above)
Comment (simon-the-shark) [reply lib/features/parkings_view/parkings_view.dart:31]: but not sure how this would look, opening up a task for the future reference
